### ISSUE TITLE
Pandas converter: add support and tests for StringDtype

### ISSUE THREAD

**Issue by krassowski** [rpy2/rpy2#669]
Pandas 1.0 [introduced a new way of storing string data](https://pandas.pydata.org/pandas-docs/stable/user_guide/text.html#text-data-types) using a dedicated `StringDtype` (the earlier implementation which uses `object` remains default for now). This does not seem to play well with the rpy2's converter currently:

```python
%load_ext rpy2.ipython
from pandas import *
characters= = Series(['a', 'b'], dtype=StringDtype())
%R -i characters
```

```python
~/.pyenv/versions/3.8.1/envs/playground/lib/python3.8/site-packages/rpy2/robjects/pandas2ri.py in py2rpy_pandasseries(obj)
    159         # current conversion as performed by numpy
    160 
--> 161         res = func(obj)
    162         if len(obj.shape) == 1:
    163             if (obj.dtype != dt_O_type):

~/.pyenv/versions/3.8.1/envs/playground/lib/python3.8/site-packages/rpy2/robjects/numpy2ri.py in numpy2rpy(o)
     81     """ Augmented conversion function, converting numpy arrays into
     82     rpy2.rinterface-level R structures. """
---> 83     if not o.dtype.isnative:
     84         raise ValueError('Cannot pass numpy arrays with non-native '
     85                          'byte orders at the moment.')

AttributeError: 'StringDtype' object has no attribute 'isnative'
```

I am not sure if this should be addressed in rpy2 or upstream (`isnative` sounds like a nice property of a dtype), but once pandas switches to this as a default this issue will likely resurface if not dealt with now.

**Comment by lgautier** [comment]
I'll look at it. Speaking of which, the fix to issue #668 in PR #672 will handle `pandas.NA` used with `StringDType`.

**Comment by krassowski** [comment]
Thank you. Please let me know if I can help with beta-testing or otherwise. I really appreciate your continuous support and developement of rpy2.

**Comment by lgautier** [comment]
Thanks for the offer. Taking the time to test things out are report all the issues as you already do is greatly appreciated.

The rewrite to use cffi has helped me pay some of the technical debt back and clear complexities that were only existing to ensure compatibility with versions of Python or R no longer supported. I also added quite a few type hints, and invested getting more continuous automation (building, testing) after the move to Github. This should make it easier for someone (else than me) to propose changes or fixes... or help continue cleaning the code.

**Comment by lgautier** [comment]
There were a bunch issues when I tried master locally. I fixed them in PR #676 (no idea why the tests were passing on the CI but not on my local instance). StringDType seems to be handled well now.
